### PR TITLE
This involves following changes related to pkgbuilder [WIP]

### PR DESCRIPTION
**Pull request by akash1808**
1. Adding puppet modules for docker and pkgbuilder.
2. Addding variables in jio-tools.yaml
3. Moving base packages required from commonservices/jenkins.pp to commonservices/tools.pp
4. Reshuffle in site.pp

### CODE REVIEW

**Comment by akash1808** [comment]
Added changes for docker proxy and user related configurations.

File: hiera/data/cloud_provider/jio-tools.yaml [hunk 1]
```diff
@@ -21,3 +21,18 @@ rjil::system::accounts::sudo_users:
...
+pkgbuilder::public_port: "9000"
+pkgbuilder::log_dir: "/var/log/pkgbuilder"
+pkgbuilder::db::db_name: "pkgbuilder"
+pkgbuilder::db::db_rootpassword: "db_password"
+pkgbuilder::db::db_username: "pkgbuilder"
+pkgbuilder::db::db_password: "change3M3"
+pkgbuilder::proxy: "%{rjil::system::apt::proxy}"
+docker::params::proxy: "%{rjil::system::apt::proxy}"
+docker::params::docker_users: ["%{pkgbuilder::username}"]
+docker::params::extra_parameters: ['--bip=192.168.1.1/24']
```
Comment (anshprat) [inline hiera/data/cloud_provider/jio-tools.yaml]: Can we reduce the use of hiera? Hiera lookups are expensive. So lets move the defaults to respective code and use hiera only for overwriting the defaults when required.

File: hiera/data/cloud_provider/jio-tools.yaml [hunk 2]
```diff
@@ -21,3 +21,18 @@ rjil::system::accounts::sudo_users:
 rjil::base::self_signed_cert: true
 
 rjil::jiocloud::consul::encrypt: "%{consul_gossip_encrypt}"
+
+pkgbuilder::install_dir: "/home/pkgbuilder"
```
Comment (anshprat) [inline hiera/data/cloud_provider/jio-tools.yaml]: /home is not a good place for installs.

**Comment by anshprat** [comment]
looks good. merging as it does not touch other parts + has been tested on tools02 to a certain extent.

**Pull request merged**